### PR TITLE
fixed #341: querying secondary indexes may yield expected results if …

### LINKED ISSUE #341: querying secondary indexes may yield expected results if tables have the same index name
Steps to reproduce:
- create two tables for a given tenant, both with a GSI that has the same name
- insert a item into each table putting the same value in the GSI's HK field
- query one of the tables using the GSI

Expected result: find the record
Actual result: you get two records, one for each table

### PR DESCRIPTION
**Pull request by msgroi**

### CODE REVIEW

File: src/main/java/com/salesforce/dynamodbv2/mt/mappers/metadata/DynamoTableDescriptionImpl.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ public DynamoSecondaryIndex findSi(String indexName) {
             si = getLsi(indexName);
         }
         if (!si.isPresent()) {
-            throw new IllegalArgumentException("secondary index '" + indexName + "' not found");
+            throw new IllegalArgumentException("secondary index '" + indexName + "' not found on " + getTableName());
```
Comment (msgroi) [inline src/main/java/com/salesforce/dynamodbv2/mt/mappers/metadata/DynamoTableDescriptionImpl.java]: Unrelated improvement to error handling

File: src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldMapper.java [hunk 2]
```diff
@@ -37,9 +38,8 @@
     AttributeValue apply(FieldMapping fieldMapping, AttributeValue unqualifiedAttribute) {
         return new AttributeValue(
             fieldPrefixFunction.apply(mtContext,
-                fieldMapping.getIndexType() == TABLE
-                    ? virtualTableName
-                    : fieldMapping.getVirtualIndexName(),
+                virtualTableName
+                + (fieldMapping.getIndexType() == SECONDARYINDEX ? SEPARATOR + fieldMapping.getVirtualIndexName() : ""),
```
Comment (msgroi) [inline src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldMapper.java]: This is the actual fix.

File: src/test/java/com/salesforce/dynamodbv2/ScanTest.java [hunk 3]
```diff
@@ -174,7 +171,7 @@ public void setupTableData(AmazonDynamoDB amazonDynamoDb, ScalarAttributeType ha
             CreateTableRequest createTableRequest) {
             int ordinal = (Integer.parseInt(org.substring(org.indexOf("-") + 1)) - 1) % ORGS_PER_TEST;
             int putCount = ordinal < orgPutCounts.size() ? orgPutCounts.get(ordinal) : new Random().nextInt(10);
-            MT_CONTEXT.setContext(org);
+            getMtContext().setContext(org);
```
Comment (msgroi) [inline src/test/java/com/salesforce/dynamodbv2/ScanTest.java]: Unrelated simplification.

File: src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldMapper.java (hunk 2 above)
Comment (busjaeger) [reply src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldMapper.java]: Why append the index name?

Comment (busjaeger) [reply src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldMapper.java]: Also, this is a breaking change, right?

Comment (msgroi) [reply src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldMapper.java]: I don't know why I included in the index name.

Comment (msgroi) [reply src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldMapper.java]: This is a breaking change for anyone who happened to use gsi's.

Comment (msgroi) [reply src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldMapper.java]: fixed in 4910add08ec10862669e0089d59a222d4602de48

**Pull request merged**